src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/NuGetTargetFrameworkMonikerComparer.cs

```diff
@@ -112,14 +112,17 @@ public bool TryMerge(TargetFrameworkMoniker tfm1, TargetFrameworkMoniker tfm2, [
                 _ => null,
             };
 
-            var newFramework = (platform, version) switch
+            var frameworkName = (maxFramework.Framework, maxFramework.Version) switch
             {
-                (string p, Version v) => new NuGetFramework(maxFramework.Framework, maxFramework.Version, p, v),
-                (string p, null) => new NuGetFramework(maxFramework.Framework, maxFramework.Version, p, new Version(0, 0, 0, 0)),
-                _ => new NuGetFramework(maxFramework.Framework, maxFramework.Version),
+                (".NETCoreApp", Version v) when v >= TargetFrameworkMoniker.Net50.FrameworkVersion => "net",
```

Comment:
should use constants from `TargetFrameworkMoniker.cs`

Reply:
This isn't necessary anymore as those string comparisons are handled in TargetFrameworkMoniker now. I've removed that.